src/app/cli/src/coda.ml

```diff
@@ -551,12 +551,20 @@ let daemon logger =
            >>| Or_error.ok
          in
          Stream.iter
-           (Async.Scheduler.long_cycles
+           (Async_kernel.Async_kernel_scheduler.(
+              long_cycles_with_context @@ t ())
               ~at_least:(sec 0.5 |> Time_ns.Span.of_span_float_round_nearest))
-           ~f:(fun span ->
+           ~f:(fun (span, ctx) ->
+             let tm = Time_ns.Span.to_string span in
+             (* see if backtraces reveal anything about long async cycles *)
+             let metadata =
+               ("tm", `String tm)
+               :: List.mapi ctx.backtrace_history ~f:(fun ndx bt ->
```

Comment:
Do you think it would be better to coalesce the backtrace into on message rather than multiple messages?

Reply:
For now, I want to keep them separate, preserving the history structure in the context. It might be that only the most recent one is useful, or some other collapsing could be useful. Let's wait to see what we in the logs.